### PR TITLE
Base64fix

### PR DESCRIPTION
**Pull request by KoviaX**
The branch was created by @Edward-Ross-Gridimp. Maybe he could add some explanation as to why this fix was necessary, and perhaps it would also be possible to expand on the seal/unseal message tests, as the behavioural changes are not reflected in the tests now.

### CODE REVIEW

**Comment by Edward-Ross-Gridimp** [comment]
@KoviaX Please note that when I used the library I copied it, stripped out the Spring parts and worked on a separate repository.  Therefore my changes here have not been compiled or tested, I also haven't modified the unit tests.

This change is necessary for the message exchange, without this change it is impossible to exchange messages.

The issue is that the reference lib’s seal method signs the message and then hex encodes it before base64 encoding it. In the unseal path, it is base64 decoding, hex decoding and then verifying the message signature. However, the UFTP (Shapeshifter) specification only requires Base64 encoding. It does not mention hex encoding at all.

I found the cause of the problem in the library where is it using the LazySodium Java library to sign the message. My change makes it so that only base64 encoding is used, as specified by UFTP (Shapeshifter).

P.S. I also have some other corrections to the existing unit tests which I could contribute back into the library later.

File: src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java
Comment (jasperat) [inline src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java:39]: You changed the expected public key format to a base64 string (this was a hex string before), this is a good change.
it's inconsistent if the private key is expected to be a hex string. So can you also change the private key?

File: src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java
Comment (jasperat) [inline src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java:34]: The import for Base64MessageEncoder is missing. Please make sure the code compiles

**Review by jasperat — CHANGES_REQUESTED** [review]
Please fix the failing check. See: https://github.com/shapeshifter/shapeshifter-library/pull/1/checks

File: src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java
Comment (Edward-Ross-Gridimp) [reply src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java:34]: Correction pushed to branch on appbeats repo (https://github.com/appbeats/uftplib/tree/base64fix).

Could you give my account write access on this branch please?

File: src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java
Comment (Edward-Ross-Gridimp) [reply src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java:39]: done.  I also included the missing change to the key generation from our internal repository.

Changes pushed to appbeats repo

File: src/test/java/eu/uftplib/service/UftpSigningServiceImplementationTest.java
Comment (Edward-Ross-Gridimp) [reply src/test/java/eu/uftplib/service/UftpSigningServiceImplementationTest.java:21]: The key here is base64 encoded, now with the missing change to remove the hex enencode from the library I believe this test will work.

I have tested the compliation locally with "./gradlew build", how do I run the unit tests locally from Gradle?

File: src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java
Comment (Edward-Ross-Gridimp) [reply src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java:35]: Changes now pushed to this branch.

File: src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java
Comment (Edward-Ross-Gridimp) [reply src/main/java/eu/uftplib/service/UftpSigningServiceImplementation.java]: pushed change to this repo branch now

File: src/test/java/eu/uftplib/service/UftpSigningServiceImplementationTest.java
Comment (Edward-Ross-Gridimp) [reply src/test/java/eu/uftplib/service/UftpSigningServiceImplementationTest.java:21]: I ran the command "./gradlew test".   There were no errors output.

~~~
:~/workspace/shapeshifter-library$ ./gradlew test

BUILD SUCCESSFUL in 2s
4 actionable tasks: 4 up-to-date
~~~

**Comment by jasperenexis** [comment]
@Edward-Ross-Gridimp can you look into the DCO check? https://github.com/shapeshifter/shapeshifter-library/pull/1/checks?check_run_id=4818873466
After you fixed it, we can merge the PR

**Comment by Edward-Ross-Gridimp** [comment]
@jasperenexis I have added signature to one commit, but managed to cause another commit without signature.

I will rebase to get rid of these commits and install the latest git so I can use the new rebase --signoff option...

**Comment by Edward-Ross-Gridimp** [comment]
@jasperenexis signoff done

**Pull request merged**
